stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt

```diff
@@ -311,16 +311,22 @@ public class ChatClient internal constructor(
         socket.removeListener(listener)
     }
 
-    @Deprecated(
-        message = "Use subscribe() on the client directly instead",
-        level = DeprecationLevel.WARNING
-    )
+    @Deprecated(message = "Use subscribe() on the client directly instead")
     public fun events(): ChatObservable {
         return socket.events()
     }
 
     public fun subscribe(
-        listener: (event: ChatEvent) -> Unit,
+        listener: ChatEventListener<ChatEvent>,
+    ): Disposable {
+        return eventsObservable.subscribe(listener = listener)
+    }
+
+    @Suppress("NEWER_VERSION_IN_SINCE_KOTLIN")
+    @Deprecated(message = "Use subscribe with ChatEventListener parameter")
+    @SinceKotlin("99999.9")
```

Comment:
Is there some way to compose / combine annotations in Kotlin without writing your own parser to unwrap / expand annotated annotations? Would be nice if we could have `@SinceKotlin("99999.9")` become something like `@ResolveLast` or something, or just be able to get rid of the string literal.

Reply:
I believe we don't have a way to achieve this at the moment. Would be great though!

Meta-annotations are possible in Spring (e.g. here's a [custom `SpringTest` annotation](https://github.com/zsmb13/site/blob/6d55b09c587152d75926d9d8b4e69361b3093957/site-backend/src/test/kotlin/co/zsmb/site/backend/setup/SpringTest.kt) I created that adds a whole lot of config to a test in one annotation), but they have their own special processing there as far as I understand.